packages/circle-demo-webapp/app/lib/memcache.ts

```diff
@@ -0,0 +1,41 @@
+import { sdk } from '~/lib/sdk';
+import { Token } from '~/lib/types';
+
+class MemCache<ReturnType> {
```

Comment:
we have already simple in memory caching here packages/circle-demo-webapp/app/lib/cache.ts, can't that be used?

Reply:
I do not think we need to invalidate the tokens cache. and it has more functionality (to load automatically if not exists)

Reply:
also, for this kind of cache, we do need to have an extra library installed - which increases the bundle size

Reply:
could you then also replace the previous use of cache with the new memcache you introduced?

Reply:
replaced